workspaces/mcp-integrations/plugins/backstage-mcp-plugin-backend/src/plugin.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  coreServices,
+  createBackendPlugin,
+} from '@backstage/backend-plugin-api';
+import {
+  CatalogService,
+  catalogServiceRef,
+} from '@backstage/plugin-catalog-node';
+import { actionsRegistryServiceRef } from '@backstage/backend-plugin-api/alpha';
+/**
+ * backstageMcpPlugin backend plugin
+ *
+ * @public
+ */
+export const backstageMcpPlugin = createBackendPlugin({
+  pluginId: 'backstage-mcp-plugin',
+  register(env) {
+    env.registerInit({
+      deps: {
+        actionsRegistry: actionsRegistryServiceRef,
+        logger: coreServices.logger,
+        httpAuth: coreServices.httpAuth,
+        httpRouter: coreServices.httpRouter,
+        catalog: catalogServiceRef,
+        auth: coreServices.auth,
+      },
+      // Sample action used in the Backstage docs: https://github.com/backstage/backstage/tree/master/plugins/mcp-actions-backend
+      async init({ actionsRegistry, catalog, auth }) {
+        actionsRegistry.register({
+          name: 'greet-user',
+          title: 'Greet User',
+          description: 'Generate a personalized greeting',
+          schema: {
+            input: z =>
+              z.object({
+                name: z.string().describe('The name of the person to greet'),
+              }),
+            output: z =>
+              z.object({
+                greeting: z.string().describe('The generated greeting'),
+              }),
+          },
+          action: async ({ input }) => ({
+            output: { greeting: `Hello ${input.name}!` },
+          }),
+        });
+        // This action is used to fetch the list of catalog entities from Backstage. It returns an array of entity names
+        actionsRegistry.register({
+          name: 'fetch-catalog-entities',
+          title: 'Fetch Catalog Entities',
+          description:
+            'Retrieve the list of catalog entities from the Backstage server.',
```

Comment:
Per what we've seen with the llama-stack playground experiment I set up for Ugo, we'll ultimately want a much more verbose description here.

Example invocations (even if there is only one way to call ... make that clear in the description.

Example output.

`oc new-app`-esque @johnmcollier :-)

Reply:
@Yhugieee (this is Ugo @johnmcollier though maybe you knew that from your git assistance with her) ^^ from John's demo in the parking lot earlier today, a possible "next assignment" that builds from your current one will be porting the catalog query tools over here.

Reply:
> Example invocations (even if there is only one way to call ... make that clear in the description.

I think based on today's call, we'll leave the descriptions as-is, and wait for us to port over the existing tools

> Also @johnmcollier we probably want to get her git ID properly registered in our org

Invite sent